### PR TITLE
feat/CO-43 reusable Dialog

### PR DESCRIPTION
**Pull request by arjun-at-roam**
## PR Area

- [x] Frontend

## Type of change

_What kind of change does this PR introduce?_

- [x] Feature

## Description

* Created new DialogOverlay module that encapsulates overlay functionalities

## Security & risks

- [ ] Adds new third party libraries
- [ ] Adds new `.env` variables
- [ ] Integrates with Third-party apis (dependency must be resilient)
- [ ] Has specific [OWASP security concerns](https://owasp.org/www-project-top-ten/) - list them: (also see [Fixing OWASP Top 10](https://nodegoat.herokuapp.com/tutorial))

## Pre-publish checklist

- [x] I have assigned myself to the ticket
- [x] I have performed a self-review of my code
- [x] I have performed relevant browser testing
- [ ] I have performed keyboard testing of new or modified interactive components

## Reviewer checklist

- [ ] I have run the code (if applicable)
- [ ] I have reviewed and run any tests (if applicable)

### CODE REVIEW

File: src/modules/DialogOverlay.vue [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    v-if="isDialogOpen"
+    id="widget-dialog"
+    :button-close-label="buttonCloseLabel"
+    @toggle-dialog="toggleDialog"
+  >
+    <template #header> <slot name="heading"></slot> </template>
+    <template #body>
+      <slot name="tabs"></slot>
+      <slot name="terms"></slot>
+      <slot name="applyNow"></slot>
```
Comment (arjun-at-roam) [inline src/modules/DialogOverlay.vue]: applyNow -> more generic name

File: src/modules/DialogOverlay.vue [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  <Dialog
+    v-if="isDialogOpen"
+    id="widget-dialog"
+    :button-close-label="buttonCloseLabel"
+    @toggle-dialog="toggleDialog"
+  >
+    <template #header> <slot name="heading"></slot> </template>
+    <template #body>
+      <slot name="tabs"></slot>
+      <slot name="terms"></slot>
```
Comment (arjun-at-roam) [inline src/modules/DialogOverlay.vue]: import tabs and accordian components in.
Pass data as props into tabs and accordian

File: src/modules/DialogOverlay.vue [hunk 3]
```diff
@@ -0,0 +1,73 @@
+<template>
+  <Dialog
+    v-if="isDialogOpen"
+    id="widget-dialog"
+    :button-close-label="buttonCloseLabel"
+    @toggle-dialog="toggleDialog"
+  >
+    <template #header> <slot name="header"></slot> </template>
```
Comment (ghost) [inline src/modules/DialogOverlay.vue]: Nit: Looks like the formatting might be a bit off here

File: src/modules/DialogOverlay.vue [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+          </Tab>
+          <Tab tab-id="18months" :active-tab-id="activeTabId">
+            <DataList :contents="tabsContents('18months')" />
+          </Tab>
+        </template>
+      </Tabs>
+      <Accordion id="widget-terms" :content="accordianData">
+        Terms & Conditions
+      </Accordion>
+      <slot name="footer"></slot>
```
Comment (ghost) [inline src/modules/DialogOverlay.vue]: Nit: This slot can probably be self closing

**Review by ghost — APPROVED** [review]

**Review by jpyepez-roam — APPROVED** [review]
Looks good! Approved

File: src/modules/DialogOverlay.vue (hunk 4 above)
Comment (arjun-at-roam) [reply src/modules/DialogOverlay.vue]: great eye!
done

File: src/modules/DialogOverlay.vue (hunk 3 above)
Comment (arjun-at-roam) [reply src/modules/DialogOverlay.vue]: thank you. done.

**Pull request merged**